From ab20b5052df6a59b0b952abecd238a6a8f84b961 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Fri, 20 Feb 2026 11:55:03 +0000
Subject: [PATCH] Remove service-virtualization from system server jars

com.android.virt is excluded from PRODUCT_PACKAGES for GSI builds
since it requires Pixel-specific pKVM firmware. Remove its system
server jar from PRODUCT_APEX_SYSTEM_SERVER_JARS to match, otherwise
the dexpreopt check fails on missing compilation artifacts.
---
 target/product/default_art_config.mk | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/target/product/default_art_config.mk b/target/product/default_art_config.mk
index 543e565..303c012 100644
--- a/target/product/default_art_config.mk
+++ b/target/product/default_art_config.mk
@@ -197,9 +197,6 @@ ifeq ($(RELEASE_ONDEVICE_INTELLIGENCE_MODULE),true)
 
 endif
 
-ifeq ($(RELEASE_AVF_ENABLE_LLPVM_CHANGES),true)
-  PRODUCT_APEX_SYSTEM_SERVER_JARS += com.android.virt:service-virtualization
-endif
 
 # Use $(wildcard) to avoid referencing the profile in thin manifests that don't have the
 # art project.
-- 
2.52.0

